File: backend/src/main/java/com/back/domain/adoption/dto/response/ApplicationResponseDto.java
package com.back.domain.adoption.dto.response;

import com.back.domain.applicant.dto.response.ApplicantResponseDto;
import com.back.domain.pet.dto.response.PetInfoResponseDto;
import com.back.domain.adoption.entity.Adoption;
import com.back.domain.adoption.enums.RequestStatus;
import com.back.domain.care.entity.Care;
import java.time.LocalDateTime;
import lombok.Builder;

@Builder
public record ApplicationResponseDto(
        Long id,
        String title,
        String type, // "ADOPTION" 또는 "CARE"
        ApplicantResponseDto applicantInfo,
        String anotherPets,
        String experience,
        String message,
        RequestStatus status,
        LocalDateTime createdAt,
        PetInfoResponseDto petInfo,
        LocalDateTime desiredStartDate, // Care인 경우에만 사용
        LocalDateTime desiredEndDate    // Care인 경우에만 사용
) {
    
    public static ApplicationResponseDto fromAdoption(Adoption adoption) {
        return ApplicationResponseDto.builder()
                .id(adoption.getId())
                .title(adoption.getTitle())
                .type("ADOPTION")
                .applicantInfo(ApplicantResponseDto.builder()
                        .id(adoption.getApplicant().getId())
                        .name(adoption.getApplicant().getName())
                        .phone(adoption.getApplicant().getPhone())
                        .email(adoption.getApplicant().getEmail())
                        .address(adoption.getApplicant().getAddress())
                        .build())
                .anotherPets(adoption.getAnotherPets())
                .experience(adoption.getExperience())
                .message(adoption.getMessage())
                .status(adoption.getStatus())
                .createdAt(adoption.getCreatedAt())
                .petInfo(PetInfoResponseDto.from(adoption.getPet()))
                .build();
    }
    
    public static ApplicationResponseDto fromCare(Care care) {
        return ApplicationResponseDto.builder()
                .id(care.getId())
                .title(care.getTitle())
                .type("CARE")
                .applicantInfo(ApplicantResponseDto.builder()
                        .id(care.getApplicant().getId())
                        .name(care.getApplicant().getName())
                        .phone(care.getApplicant().getPhone())
                        .email(care.getApplicant().getEmail())
                        .address(care.getApplicant().getAddress())
                        .build())
                .anotherPets(care.getAnotherPets())
                .experience(care.getExperience())
                .message(care.getMessage())
                .status(care.getStatus())
                .createdAt(care.getCreatedAt())
                .petInfo(PetInfoResponseDto.from(care.getPet()))
                .desiredStartDate(care.getDesiredStartDate())
                .desiredEndDate(care.getDesiredEndDate())
                .build();
    }
} 

File: backend/src/main/java/com/back/domain/adoption/dto/response/ApplicationResponseDto.kt
package com.back.domain.adoption.dto.response

import com.back.domain.adoption.entity.Adoption
import com.back.domain.adoption.enums.RequestStatus
import com.back.domain.applicant.dto.response.ApplicantResponseDto
import com.back.domain.care.entity.Care
import com.back.domain.pet.dto.response.PetInfoResponseDto
import java.time.LocalDateTime

data class ApplicationResponseDto(
    val id: Long?,
    val title: String,
    val type: String,  // "ADOPTION" 또는 "CARE"
    val applicantInfo: ApplicantResponseDto,
    val anotherPets: String?,
    val experience: String?,
    val message: String?,
    val status: RequestStatus,
    val createdAt: LocalDateTime?,
    val petInfo: PetInfoResponseDto,
    val desiredStartDate: LocalDateTime?,  // Care인 경우에만 사용
    val desiredEndDate: LocalDateTime? // Care인 경우에만 사용
) {
    companion object {
        fun fromAdoption(adoption: Adoption): ApplicationResponseDto? {
            return ApplicationResponseDto(
                id = adoption.id,
                title = adoption.title,
                type = "ADOPTION",
                applicantInfo = ApplicantResponseDto(
                    id = adoption.applicant.id,
                    name = adoption.applicant.name,
                    phone = adoption.applicant.phone,
                    email = adoption.applicant.email,
                    address = adoption.applicant.address
                ),
                anotherPets = adoption.anotherPets,
                experience = adoption.experience,
                message = adoption.message,
                status = adoption.status,
                createdAt = adoption.createdAt,
                petInfo = PetInfoResponseDto.from(adoption.pet),
                desiredStartDate = null,
                desiredEndDate = null
            )
        }

        fun fromCare(care: Care): ApplicationResponseDto? {
            return ApplicationResponseDto(
                id = care.id,
                title = care.title,
                type = "CARE",
                applicantInfo = ApplicantResponseDto(
                    id = care.applicant.id,
                    name = care.applicant.name,
                    phone = care.applicant.phone,
                    email = care.applicant.email,
                    address = care.applicant.address
                ),
                anotherPets = care.anotherPets,
                experience = care.experience,
                message = care.message,
                status = care.status,
                createdAt = care.createdAt,
                petInfo = PetInfoResponseDto.from(care.pet),
                desiredStartDate = care.desiredStartDate,
                desiredEndDate = care.desiredEndDate
            )
        }
    }
}